kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -368,22 +372,128 @@ def _document(cls):
             n=cls._doc_array_n)
 
 
-class TimeFrequencyScatteringBase(ScatteringBase1D):
+class TimeFrequencyScatteringBase():
+    def __init__(self, FrontendClass, J_fr=None, Q_fr=1, oversampling_fr='auto'):
+        self.FrontendClass = FrontendClass
+        self.J_fr = J_fr
+        self.Q_fr = Q_fr
+        self.oversampling_fr = oversampling_fr
+
+    def build(self):
+        # First-order scattering object for the frequency variable
+        self.max_order_fr = 1
+        self.shape_fr = self.get_shape_fr()
+        # Calibrate all filters with respect to the most obtainable freq rows
+        self.N_fr = self.shape_fr[-1]
+        if self.J_fr is None:
+            self.J_fr = self.get_J_fr()
+
+        self.sc_freq = self.FrontendClass(
+            self.J_fr, self.N_fr, self.Q_fr, self.max_order_fr, self.average,
+            self.oversampling, self.vectorize, self.out_type, self.backend)
+
+        self.compute_padding_fr()
+        self.sc_freq.psi1_f = resample_frequential_filters(
+            **self.get_fr_params('psi1_f', 'J_pad', 'normalize', 'P_max', 'eps'))
+
+        # build spin up & down wavelets
+        # TODO make `create_filters`?
+        self.sc_freq.psi1_f_up = self.sc_freq.psi1_f
+        self.sc_freq.psi1_f_down = compute_spin_down_filters(
+            self.sc_freq.psi1_f_up, self.backend)
+        del self.sc_freq.psi1_f
+
+    def compute_padding_fr(self):
+        """Long story"""  # TODO
```

Comment:
or else you can rename the method to `_compute_padding_fr` and leave it undocumented ;p

Reply:
i don't think it'll ever serve in user code

Reply:
To contrary, here we should explain much of the circus in `_frequency_scattering` and `_joint_lowpass`. (or elsewhere, but it must be explained)